challenge_receptionist/src/challenge_receptionist/find_empty_seat.py

```diff
@@ -46,77 +128,79 @@ class FindEmptySeat(smach.StateMachine):
     def __init__(self, robot, seats_to_inspect, room, seat_is_for=None):
         smach.StateMachine.__init__(self, outcomes=['succeeded', 'failed'])
 
+        people = ds.VariableDesignator(resolve_type=[Entity])
+
         seats = SeatsInRoomDesignator(robot, seats_to_inspect, room, "seats_in_room")
-        seat_ent_des = ds.VariableDesignator(resolve_type=Entity)
+        seat_ent_des = ds.VariableDesignator(resolve_type=Entity, name='seat_entity_designator')
         if seat_is_for:
             ds.check_type(seat_is_for, str)
         else:
-            seat_is_for = ds.Designator(' ')
+            seat_is_for = ds.Designator('someone')
+
+        seat_volume_des = ds.VariableDesignator(resolve_type=str, name='seat_volume_name_designator')
 
         with self:
             smach.StateMachine.add('SAY_LETS_FIND_SEAT',
-                                   states.SayFormatted(robot,
-                                                       ["Let me find a place for {name} to sit. Please be patient while I check out where there's place to sit"],
-                                                       name=seat_is_for,
-                                                       block=False),
-                                   transitions={'spoken': 'ITERATE_NEXT_SEAT'})
-
-            smach.StateMachine.add('ITERATE_NEXT_SEAT',
-                                   states.IterateDesignator(seats, seat_ent_des.writeable),
-                                   transitions={'next': 'CHECK_SEAT_EMPTY',
-                                                'stop_iteration': 'SAY_NO_EMPTY_SEATS'})
```

Comment:
By removing the iteration over the different seats you assume that you will find all people in the room in one go. I am not saying that it's a bad assumption, just be aware that you made it

Reply:
True. The state for finding ppl in room takes care of looking around.